app/src/main/kotlin/it/vfsfitvnm/vimusic/service/PlayerService.kt

```diff
@@ -1136,7 +1139,12 @@ class PlayerService : InvincibleService(), Player.Listener, PlaybackStatsListene
 
                         if (body?.videoDetails?.videoId != videoId) throw VideoIdMismatchException()
 
-                        val format = body.streamingData?.highestQualityFormat
+                        val format = if (PlayerPreferences.highQualityAudio) {
```

Comment:
I think it does not make sense to use the lowest quality format, or we should at least indicate this clearly to the user, since if we are going to use this code, all songs played with the high quality setting off are forever going to be in low quality from that point forward. I think we should differentiate cache keys in order to achieve consistent behavior.

Reply:
It's not mean to only use the lowest quality, its because the more quality the song has, the more size it gets stored in disk.
Low Quality Audio and disabling Cache are meant for users that only want to play the song online or store the song in disk but in low size.

Reply:
I think the user should be able to choose priority over formats themselves though, using a list like [this one](https://gist.github.com/AgentOak/34d47c65b1d28829bb17c24c04a0096f)